### PR TITLE
Tokenizing TabBar

### PR DESCRIPTION
**Pull request by edjamesmsft**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Add Tokens support for the UIKit control TabBarView and TabBarItem. Not included in this PR is SideTabBar or BadgeLabel, which will come in another change.

### Verification

Tested the TabBarView demo on multiple device types, dark and light mode, differing font size.

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| TokenizeTabBarBefore | TokenizeTabBarAfter2 |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [x] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [x] Different resolutions (1x, 2x, 3x)
- [x] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/970)

### CODE REVIEW

File: ios/FluentUI/Tab Bar/TabBarTokens.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+/// Design token set for the `TabBar`.
+open class TabBarTokens: ControlTokens {
+    /// The height of the `TabBar` when in portrait view on a phone
+    open var phonePortraitHeight: CGFloat { 48.0 }
+
+    /// The height of the `TabBar` when in landscape view on a phone
+    open var phoneLandscapeHeight: CGFloat { 40.0 }
+
+    /// The height of the `TabBar` when on a non-phone device
+    open var padHeight: CGFloat { 48.0 }
```
Comment (harrieshin) [inline ios/FluentUI/Tab Bar/TabBarTokens.swift:17]: even in compact mode of iPad we are always 48?

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift [hunk 2]
```diff
@@ -193,3 +193,59 @@ extension TabBarViewDemoController: TabBarViewDelegate {
...
+        _ = tabBarView?.overrideTokens(tokens)
+    }
+
+    func isThemeWideOverrideApplied() -> Bool {
+        return self.view.window?.fluentTheme.tokenOverride(for: TabBarView.self) != nil
+    }
+
+    // MARK: - Custom tokens
+    private class ThemeWideOverrideTabBarTokens: TabBarTokens {
+        override var badgeHeight: CGFloat {
```
Comment (harrieshin) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift:225]: why is badgeheight part of themewideoverridetabbartoken?

File: ios/FluentUI.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 3]
```diff
@@ -6,17 +6,17 @@
         "repositoryURL": "https://github.com/microsoft/appcenter-sdk-apple.git",
         "state": {
           "branch": null,
-          "revision": "25f64229373de97ff3920941cd52203193e5d8be",
-          "version": "4.3.0"
+          "revision": "8354a50fe01a7e54e196d3b5493b5ab53dd5866a",
+          "version": "4.4.2"
```
Comment (huwilkes) [inline ios/FluentUI.xcworkspace/xcshareddata/swiftpm/Package.resolved:10]: Were these changes intentional?

File: ios/FluentUI/Tab Bar/TabBarItemView.swift [hunk 4]
```diff
@@ -238,7 +259,7 @@ class TabBarItemView: UIControl {
         if let window = window {
             let primaryColor = Colors.primary(for: window)
```
Comment (huwilkes) [inline ios/FluentUI/Tab Bar/TabBarItemView.swift:260]: You might want to grab tokens.aliasTokens.foreground[.brandRest] instead of Colors.primary(for: window), though I think it's more likely that this should just be a tokenized value?

File: ios/FluentUI/Tab Bar/TabBarItemView.swift [hunk 5]
```diff
@@ -103,6 +130,11 @@ class TabBarItemView: UIControl {
 			container.widthAnchor.constraint(lessThanOrEqualTo: widthAnchor)
 		])
 
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(themeDidChange),
+                                               name: .didChangeTheme,
+                                               object: nil)
```
Comment (huwilkes) [inline ios/FluentUI/Tab Bar/TabBarItemView.swift:136]: If the TabBarView is in charge of updating the TabBarItemView's tokens, I'm not sure you need this observer on both. I think having the TabBarItemView update all the tokenized values after the TabBarView sets the new tokens should be enough.

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift [hunk 6]
```diff
@@ -193,3 +193,59 @@ extension TabBarViewDemoController: TabBarViewDelegate {
...
+                return ThemeWideOverrideTabBarTokens()
+            }
+        }
+
+        fluentTheme.register(controlType: TabBarView.self, tokens: tokensClosure)
+    }
+
+    func perControlOverrideDidChange(isOverrideEnabled: Bool) {
+        let tokens = (isOverrideEnabled ? PerControlOverrideTabBarItemTokens() : nil)
+        _ = tabBarView?.overrideTokens(tokens)
```
Comment (huwilkes) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift:216]: Probably outside the scope of this PR, but it's strange to me that the tabBarView of the TabBarViewDemo is optional?

File: ios/FluentUI/Tab Bar/TabBarItemView.swift (hunk 4 above)
Comment (huwilkes) [reply ios/FluentUI/Tab Bar/TabBarItemView.swift:260]: For some reason I forgot about globalTokens.brandColors[.primary], but that's a better option.

File: ios/FluentUI/Tab Bar/TabBarTokens.swift [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+                            lightHighContrast: ColorValue(0x303030) /* gray700 */,
+                            dark: ColorValue(0x919191) /* gray400 */,
+                            darkHighContrast: ColorValue(0xC8C8C8) /* gray200 */)
+    }
+
+    /// The vertical spacing of the `TabBarItem` within the TabBar
+    open var spacingVertical: CGFloat { 3.0 }
+
+    /// The horizontal spacing of the `TabBarItem` within the TabBar
+    open var spacingHorizontal: CGFloat { 8.0 }
```
Comment (huwilkes) [inline ios/FluentUI/Tab Bar/TabBarTokens.swift:31]: Could we use globalTokens.spacing[.xSmall]?

File: ios/FluentUI/Tab Bar/TabBarTokens.swift [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+    /// The size of the image associated with the `TabBarItem` when the device is in portrait mode
+    open var portraitImageSize: CGFloat { 28.0 }
```
Comment (huwilkes) [inline ios/FluentUI/Tab Bar/TabBarTokens.swift:34]: Could we use globalTokens.iconSize[.large]?

File: ios/FluentUI/Tab Bar/TabBarTokens.swift [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+    /// The size of the image associated with the `TabBarItem` when the device is in portrait mode and has a label
+    open var portraitImageWithLabelSize: CGFloat { 24.0 }
```
Comment (huwilkes) [inline ios/FluentUI/Tab Bar/TabBarTokens.swift:37]: Could we use globalTokens.iconSize[.medium]?

File: ios/FluentUI/Tab Bar/TabBarItemView.swift [hunk 10]
```diff
@@ -248,21 +269,21 @@ class TabBarItemView: UIControl {
 
         if isInPortraitMode {
             container.axis = .vertical
-            container.spacing = Constants.spacingVertical
+            container.spacing = tokens.spacingVertical
             titleLabel.style = .button2
             titleLabel.maxFontSize = 10
```
Comment (huwilkes) [inline ios/FluentUI/Tab Bar/TabBarItemView.swift:274]: `titleLabel.style` and `.maxFontSize` should probably be tokenized as one FontInfo token

File: ios/FluentUI/Tab Bar/TabBarView.swift [hunk 11]
```diff
@@ -90,7 +95,7 @@ open class TabBarView: UIView {
         topBorderLine.translatesAutoresizingMaskIntoConstraints = false
         addSubview(topBorderLine)
 
-        heightConstraint = stackView.heightAnchor.constraint(equalToConstant: traitCollection.userInterfaceIdiom == .phone ? Constants.phonePortraitHeight : Constants.padHeight)
+        heightConstraint = stackView.heightAnchor.constraint(equalToConstant: traitCollection.userInterfaceIdiom == .phone ? tokens.phonePortraitHeight : tokens.padHeight)
```
Comment (rdeassis) [inline ios/FluentUI/Tab Bar/TabBarView.swift:98]: I think you can make this a non-optional lazy var and initialize it with the contents like this:

```
private lazy var heightConstraint: NSLayoutConstraint = {
        stackView.heightAnchor.constraint(equalToConstant: traitCollection.userInterfaceIdiom == .phone ? tokens.phonePortraitHeight : tokens.padHeight)
    }()
```

This will avoid the force unwrap (!) in line 99 below.

File: ios/FluentUI.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 12]
```diff
@@ -6,17 +6,17 @@
         "repositoryURL": "https://github.com/microsoft/appcenter-sdk-apple.git",
         "state": {
           "branch": null,
-          "revision": "25f64229373de97ff3920941cd52203193e5d8be",
```
Comment (rdeassis) [inline ios/FluentUI.xcworkspace/xcshareddata/swiftpm/Package.resolved:9]: The changes in this file don't seem to be part of the PR.
Do you have your branch up to date with the vnext-tokens branch?

File: ios/FluentUI/Tab Bar/TabBarTokens.swift (hunk 1 above)
Comment (mischreiber) [reply ios/FluentUI/Tab Bar/TabBarTokens.swift:17]: Curious, would you expect it shorter or taller? I think that matches the phone portrait height (and looks like that was the case before tokenizing too); only phone+landscape is shorter.

File: ios/FluentUI/Tab Bar/TabBarTokens.swift [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+
+    /// The size of the image associated with the `TabBarItem` when the device is in landscape mode
+    open var landscapeImageSize: CGFloat { 24.0 }
+
+    /// The vertical offset of the `Badge` associated with this `TabBarItem`
+    open var badgeVerticalOffset: CGFloat { -4.0 }
```
[the remark this answers is not in the document]
Comment (mischreiber) [reply ios/FluentUI/Tab Bar/TabBarTokens.swift:43]: Actually I don't think I'd want to use spacing tokens for the offset values, since they're kinda two different concepts.

File: ios/FluentUI/Tab Bar/TabBarTokens.swift [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+    open var badgePortraitTitleVerticalOffset: CGFloat { -2.0 }
+
+    /// The horizontal offset of the `BadgeLabel` associated with this `TabBarItem` when the bade value is a single digit
+    open var singleDigitBadgeHorizontalOffset: CGFloat { 14.0 }
+
+    /// The horizontal offset of the `BadgeLabel` associated with this `TabBarItem` when the bade value is multiple digits
+    open var multiDigitBadgeHorizontalOffset: CGFloat { 12.0 }
+
+    /// The  height of the `BadgeLabel` associated with this `TabBarItem`
+    open var badgeHeight: CGFloat { 16.0 }
```
[the remark this answers is not in the document]
Comment (mischreiber) [reply ios/FluentUI/Tab Bar/TabBarTokens.swift:55]: We shouldn't try to force tokens into places that they aren't made for. Themes might change these token values, and we wouldn't want unexpected consequences in other parts of the app.

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift [hunk 15]
```diff
@@ -193,3 +193,59 @@ extension TabBarViewDemoController: TabBarViewDelegate {
...
+    private class ThemeWideOverrideTabBarTokens: TabBarTokens {
+        override var badgeHeight: CGFloat {
+            return .init(CGFloat(40.0))
+        }
+    }
+
+    private class PerControlOverrideTabBarItemTokens: TabBarTokens {
+        override var unselectedColor: DynamicColor {
+            return .init(
+                         light: globalTokens.sharedColors[.cornflower][.tint20], // ColorValue(0x6E6E6E) /* gray500 */,
```
Comment (mischreiber) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift:233]: Nit: the `light:` tag should be on the same line as the `.init(` here

File: ios/FluentUI.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 12 above)
Comment (mischreiber) [reply ios/FluentUI.xcworkspace/xcshareddata/swiftpm/Package.resolved:9]: Looks like Xcode decided to bump the versions of the demo app's dependencies. Can probably revert this file for now, or maybe just leave it--not sure how/when we update these normally anyway.

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift [hunk 16]
```diff
@@ -193,3 +193,59 @@ extension TabBarViewDemoController: TabBarViewDelegate {
...
+                         lightHighContrast: globalTokens.sharedColors[.cornflower][.tint40], // ColorValue(0x303030) /* gray700 */,
+                         dark: globalTokens.sharedColors[.teal][.tint40], // ColorValue(0x919191) /* gray400 */,
+                         darkHighContrast: globalTokens.sharedColors[.darkTeal][.tint40] /* gray200 */)
```
Comment (mischreiber) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift:236]: Did you mean to leave the commented values in?

File: ios/FluentUI/Tab Bar/TabBarItemView.swift (hunk 4 above)
Comment (mischreiber) [reply ios/FluentUI/Tab Bar/TabBarItemView.swift:260]: First off, yeah this should definitely be a token as well (`selectedColor` perhaps?). 

Second, agreed that we should use alias tokens rather than global tokens whenever possible.

File: ios/FluentUI/Tab Bar/TabBarView.swift (hunk 11 above)
Comment (mischreiber) [reply ios/FluentUI/Tab Bar/TabBarView.swift:98]: Question for @rdeassis: why does it need the `{ .. }()`? Could that line be more simply written as 
```swift
private lazy var heightConstraint: NSLayoutConstraint = stackView.heightAnchor.constraint(equalToConstant: traitCollection.userInterfaceIdiom == .phone ? tokens.phonePortraitHeight : tokens.padHeight)
```

File: ios/FluentUI/Tab Bar/TabBarTokens.swift (hunk 1 above)
Comment (edjamesmsft) [reply ios/FluentUI/Tab Bar/TabBarTokens.swift:17]: I was operating under the assumption that we were trying not to change any of the look and feel of the controls. If I misunderstood, I'm happy to reference whatever is the correct Figma file and update the value.

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift (hunk 2 above)
Comment (edjamesmsft) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift:225]: I will change this to tint color.

File: ios/FluentUI.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 3 above)
Comment (edjamesmsft) [reply ios/FluentUI.xcworkspace/xcshareddata/swiftpm/Package.resolved:10]: They were not. What is the general policy for automatic changes like this?

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift (hunk 6 above)
Comment (edjamesmsft) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift:216]: Definitely outside the scope of this PR, though there are other demo issues I'd like to clear up (like the leading badge not disappearing after toggling badges off), so I will try to tackle this issue in another change.

File: ios/FluentUI/Tab Bar/TabBarTokens.swift (hunk 13 above)
[the remark this answers is not in the document]
Comment (edjamesmsft) [reply ios/FluentUI/Tab Bar/TabBarTokens.swift:43]: I'd be inclined to agree with Mike - makes sense for the sizes, maybe not as much for offsets (what is an xxsmall offset?).

File: ios/FluentUI/Tab Bar/TabBarTokens.swift (hunk 8 above)
Comment (edjamesmsft) [reply ios/FluentUI/Tab Bar/TabBarTokens.swift:34]: I know they map to the same value currently, but the intent is not the same (this is a spacing, not an icon size). I would worry that if we ever change iconSize[.large], it wouldn't take this scenario into account.

File: ios/FluentUI/Tab Bar/TabBarTokens.swift [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+    open var badgeVerticalOffset: CGFloat { -4.0 }
+
+    /// The vertical offset of the `BadgeLabel` associated with this `TabBarItem` when the device is in portrait mode
+    open var badgePortraitTitleVerticalOffset: CGFloat { -2.0 }
+
+    /// The horizontal offset of the `BadgeLabel` associated with this `TabBarItem` when the bade value is a single digit
+    open var singleDigitBadgeHorizontalOffset: CGFloat { 14.0 }
+
+    /// The horizontal offset of the `BadgeLabel` associated with this `TabBarItem` when the bade value is multiple digits
+    open var multiDigitBadgeHorizontalOffset: CGFloat { 12.0 }
```
[the remark this answers is not in the document]
Comment (edjamesmsft) [reply ios/FluentUI/Tab Bar/TabBarTokens.swift:52]: .small maps to 12, which would change the offset here. If folks strongly agree with locking to a token, despite changing value, I can.

File: ios/FluentUI/Tab Bar/TabBarTokens.swift [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+
+    /// The  height of the `BadgeLabel` associated with this `TabBarItem`
+    open var badgeHeight: CGFloat { 16.0 }
+
+    /// The minimum width of the `BadgeLabel` associated with this `TabBarItem`
+    open var badgeMinWidth: CGFloat { 16.0 }
```
[the remark this answers is not in the document]
Comment (edjamesmsft) [reply ios/FluentUI/Tab Bar/TabBarTokens.swift:58]: They are the same, but I could easily see them being set to different values (esp as one of them is a min value, not an absolute value).

File: ios/FluentUI/Tab Bar/TabBarView.swift [hunk 19]
```diff
@@ -170,4 +177,41 @@ open class TabBarView: UIView {
...
+
+    private func updateTabBarTokens() {
+        self.tokens = resolvedTokens
+
+        self.items.forEach({ item in
+            let tokens = resolvedTokens
+            if let index = items.firstIndex(of: item) {
+                let arrangedSubviews = stackView.arrangedSubviews
+
+                if arrangedSubviews.count > index {
```
[the remark this answers is not in the document]
Comment (edjamesmsft) [reply ios/FluentUI/Tab Bar/TabBarView.swift:202]: I think combining the item and item view is a bigger change than I want to make right now, but I can try to make this a little less weird (I also thought this pattern was strange, but I don't know enough yet to know a better way to access the subviews - I'll experiment with your suggestion and see what I can do).

File: ios/FluentUI/Tab Bar/TabBarTokens.swift (hunk 8 above)
Comment (huwilkes) [reply ios/FluentUI/Tab Bar/TabBarTokens.swift:34]: This comment was about portraitImageSize, not spacingHorizontal. GitHub's comments are a bit confusing since they also show the three lines before.

File: ios/FluentUI.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 12 above)
Comment (edjamesmsft) [reply ios/FluentUI.xcworkspace/xcshareddata/swiftpm/Package.resolved:9]: Yeah I had that question previously to Harrie - what is the standard practice when dealing with automatic changes like this? It seems like they'll have to happen at some point, but I can always revert if that is preferred.

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift (hunk 16 above)
Comment (edjamesmsft) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift:236]: I did not :)

File: ios/FluentUI/Tab Bar/TabBarTokens.swift (hunk 18 above)
[the remark this answers is not in the document]
Comment (huwilkes) [reply ios/FluentUI/Tab Bar/TabBarTokens.swift:63]: Ah, sorry, I meant more of "Same comment as badgeHeight"

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift [hunk 20]
```diff
@@ -193,3 +193,70 @@ extension TabBarViewDemoController: TabBarViewDelegate {
...
+
+    func isThemeWideOverrideApplied() -> Bool {
+        return self.view.window?.fluentTheme.tokenOverride(for: TabBarView.self) != nil
+    }
+
+    // MARK: - Custom tokens
+    private class ThemeWideOverrideTabBarTokens: TabBarTokens {
+        override var selectedColor: DynamicColor {
+            return .init(
+                         light: globalTokens.sharedColors[.burgundy][.tint10],
```
Comment (huwilkes) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift:227]: I think light should be on the same line as the init

File: ios/FluentUI.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 3 above)
Comment (huwilkes) [reply ios/FluentUI.xcworkspace/xcshareddata/swiftpm/Package.resolved:10]: I think it should probably be removed from this PR, and if we think we actually need the change it can be added with a separate PR.

File: ios/FluentUI/Tab Bar/TabBarTokens.swift (hunk 17 above)
[the remark this answers is not in the document]
Comment (edjamesmsft) [reply ios/FluentUI/Tab Bar/TabBarTokens.swift:57]: Again I got messed up by github's display... 12 is correct. :)

File: ios/FluentUI/Tab Bar/TabBarTokens.swift (hunk 18 above)
[the remark this answers is not in the document]
Comment (edjamesmsft) [reply ios/FluentUI/Tab Bar/TabBarTokens.swift:63]: Got it! Then the same concern about not-quite-appropriate tokens applies here as well.

File: ios/FluentUI/Tab Bar/TabBarItemView.swift [hunk 21]
```diff
@@ -248,21 +260,21 @@ class TabBarItemView: UIControl {
...
+                suggestImageSize = titleLabel.isHidden ? tokens.portraitImageSize : tokens.portraitImageWithLabelSize
             }
         } else {
             container.axis = .horizontal
-            container.spacing = Constants.spacingHorizontal
-            titleLabel.style = .footnote
-            titleLabel.maxFontSize = 13
+            container.spacing = tokens.spacingHorizontal
+            titleLabel.style = tokens.titleLabelLandscape.0
+            titleLabel.maxFontSize = tokens.titleLabelLandscape.1.size
```
Comment (huwilkes) [inline ios/FluentUI/Tab Bar/TabBarItemView.swift:274]: I think this should be `titleLabel.font = UIFont.fluent(tokens.titleLabelLandscape, shouldScale: false)`, and then you can simplify the token to just the font info

File: ios/FluentUI/Tab Bar/TabBarTokens.swift [hunk 22]
```diff
@@ -0,0 +1,82 @@
...
+
+    /// The height of the `TabBar` when in landscape view on a phone
+    open var phoneLandscapeHeight: CGFloat { 40.0 }
+
+    /// The height of the `TabBar` when on a non-phone device
+    open var padHeight: CGFloat { 48.0 }
+
+    /// Defines the background color of the  of the `TabBarItem` when selected.
+    open var selectedColor: DynamicColor {
+        return self.aliasTokens.foregroundColors[.brandRest]
```
Comment (huwilkes) [inline ios/FluentUI/Tab Bar/TabBarTokens.swift:21]: I don't think you need the self. here

File: ios/FluentUI/Tab Bar/TabBarTokens.swift [hunk 23]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    /// Defines the background color of the  of the `TabBarItem` when not selected.
+    open var unselectedColor: DynamicColor {
+        DynamicColor(light: ColorValue(0x6E6E6E) /* gray500 */,
+                            lightHighContrast: ColorValue(0x303030) /* gray700 */,
+                            dark: ColorValue(0x919191) /* gray400 */,
+                            darkHighContrast: ColorValue(0xC8C8C8) /* gray200 */)
```
Comment (huwilkes) [inline ios/FluentUI/Tab Bar/TabBarTokens.swift:29]: Looks like the spacing on the values other than light is a bit odd. Hitting control + i should align them.

**Review by huwilkes — APPROVED** [review]

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift [hunk 24]
```diff
@@ -193,3 +193,72 @@ extension TabBarViewDemoController: TabBarViewDelegate {
...
+            return .init(light: globalTokens.sharedColors[.darkTeal][.tint20],
+                         lightHighContrast: globalTokens.sharedColors[.teal][.tint40],
+                         dark: globalTokens.sharedColors[.pumpkin][.tint40],
+                         darkHighContrast: globalTokens.sharedColors[.burgundy][.tint40])
+        }
+    }
+
+    private class PerControlOverrideTabBarItemTokens: TabBarTokens {
+        override var portraitImageSize: CGFloat {
+            return CGFloat(14.0)
```
Comment (mischreiber) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/TabBarViewDemoController.swift:241]: Minor nit: could all of these `CGFloat(n)` lines just be `n` (e.g. `14.0`)? I'd expect that Swift would automatically interpret the value as a CGFloat when returning.

File: ios/FluentUI/Tab Bar/TabBarItemView.swift [hunk 25]
```diff
@@ -143,25 +175,13 @@ class TabBarItemView: UIControl {
...
-        static let portraitImageSize: CGFloat = 28
-        static let portraitImageWithLabelSize: CGFloat = 24
-        static let landscapeImageSize: CGFloat = 24
-        static let badgeVerticalOffset: CGFloat = -4
-        static let badgePortraitTitleVerticalOffset: CGFloat = -2
-        static let singleDigitBadgeHorizontalOffset: CGFloat = 14
-        static let multiDigitBadgeHorizontalOffset: CGFloat = 12
-        static let badgeHeight: CGFloat = 16
-        static let badgeMinWidth: CGFloat = 16
         static let defaultBadgeMaxWidth: CGFloat = 42
```
[the remark this answers is not in the document]
Comment (mischreiber) [reply ios/FluentUI/Tab Bar/TabBarItemView.swift:178]: Should now be safe to remove this value, and thus the entire `Constants` struct!

File: ios/FluentUI/Tab Bar/TabBarTokens.swift (hunk 7 above)
Comment (mischreiber) [reply ios/FluentUI/Tab Bar/TabBarTokens.swift:36]: Looks like this hasn't been updated?

File: ios/FluentUI/Tab Bar/TabBarTokens.swift [hunk 26]
```diff
@@ -0,0 +1,71 @@
...
+    open var spacingHorizontal: CGFloat { 8.0 }
+
+    /// The size of the image associated with the `TabBarItem` when the device is in portrait mode
+    open var portraitImageSize: CGFloat { 28.0 }
+
+    /// The size of the image associated with the `TabBarItem` when the device is in portrait mode and has a label
+    open var portraitImageWithLabelSize: CGFloat { 24.0 }
+
+    /// The size of the image associated with the `TabBarItem` when the device is in landscape mode
+    open var landscapeImageSize: CGFloat { 24.0 }
```
[the remark this answers is not in the document]
Comment (mischreiber) [reply ios/FluentUI/Tab Bar/TabBarTokens.swift:45]: Same here.

File: ios/FluentUI/Tab Bar/TabBarTokens.swift [hunk 27]
```diff
@@ -0,0 +1,71 @@
...
+
+    /// The vertical offset of the `Badge` associated with this `TabBarItem`
+    open var badgeVerticalOffset: CGFloat { -4.0 }
+
+    /// The vertical offset of the `BadgeLabel` associated with this `TabBarItem` when the device is in portrait mode
+    open var badgePortraitTitleVerticalOffset: CGFloat { -2.0 }
```
[the remark this answers is not in the document]
Comment (mischreiber) [reply ios/FluentUI/Tab Bar/TabBarTokens.swift:51]: Another?

File: ios/FluentUI/Tab Bar/TabBarView.swift (hunk 11 above)
Comment (mischreiber) [reply ios/FluentUI/Tab Bar/TabBarView.swift:98]: Since it's been initialized at declaration time, we shouldn't need an initialization here as well.

**Review by mischreiber — APPROVED** [review]
Signed off assuming we switch `landscapeImageSize` to use `globalTokens.iconSize[.medium]` as discussed offline!
